src/main/java/com/google/api/pathtemplate/PathTemplate.java

```diff
@@ -1015,6 +1013,20 @@ public String encode(String... values) {
     return builder.build();
   }
 
+  private static boolean isSegmentBeginOrEndInvalid(String seg) {
+    if (seg.length() == 1 && COMPLEX_DELIMITER_PATTERN.matcher(seg).find()) {
+      return true;
+    }
+    //A segment like .{well}-{known} or {well}-{known}. is invalid.
+    //A segment like .well-known or .well-{known} will be considered a literal hence is valid
+    return (COMPLEX_DELIMITER_PATTERN.matcher(seg.substring(0, 1)).find()
+            && seg.length() > 1
+            && seg.charAt(1) == '{')
+        || (COMPLEX_DELIMITER_PATTERN.matcher(seg.substring(seg.length() - 1)).find()
+            && seg.length() > 1
```

Comment:
`seg.length() > 1` is redundantly mentioned in both portions of the OR statement.

Reply:
Yes, you are right, tested without `seg.length() > 1` and all test cases passed, will remove it.